Sources/PubNub/Helpers/Crypto/Miscellaneous/CryptoInputStream.swift

```diff
@@ -57,38 +76,63 @@ public class CryptoInputStream: InputStream {
   private weak var _delegate: StreamDelegate?
   private var _streamStatus: Stream.Status = .notOpen
   private var _streamError: Error?
-
-  public init(_ operation: Crypto.Operation, input: InputStream, contentLength: Int, with crypto: Crypto) {
+  
+  // A flag describing whether an IV vector is included at the beginning of encoded/decoded content
+  private let includeInitializationVectorInContent: Bool
+
+  public init(
+    operation: CryptoInputStream.Operation,
+    input: InputStream,
+    contentLength: Int,
+    with crypto: CryptoInputStream.DataSource,
+    includeInitializationVectorInContent: Bool = false
+  ) {
     self.operation = operation
     self.crypto = crypto
-    // We should always be using a random IV
-    self.crypto.randomizeIV = true
-
-    rawDataLength = contentLength - (operation == .decrypt ? crypto.cipher.blockSize : 0)
-    // The estimated content length is the IV length plus the crypted length
-    estimatedCryptoCount = crypto.cipher.blockSize + crypto.cipher.outputSize(from: rawDataLength)
+    self.includeInitializationVectorInContent = includeInitializationVectorInContent
+    
+    if includeInitializationVectorInContent {
+      rawDataLength = contentLength - (operation == .decrypt ? crypto.cipher.blockSize : 0)
+      // The estimated content length is the IV length plus the crypted length
+      estimatedCryptoCount = crypto.cipher.blockSize + crypto.cipher.outputSize(from: rawDataLength)
```

Comment:
As mentioned above, `CCCrypto` can help with estimated output size computation.

Reply:
I moved it from the current master implementation that's available [here](https://github.com/pubnub/swift/blob/master/Sources/PubNub/Helpers/Crypto/Crypto.swift#L142-L147). The reason I moved this code was not to alter/modify `CryptoStream` or `CryptoInputStream` already present before the more it's necessary. 

[CryptoStream](https://github.com/pubnub/swift/blob/master/Sources/PubNub/Helpers/Crypto/CryptoStream.swift) looks like a wrapper for Swift's CommonCrypto operations. In general, I agree with you it's fine to use CommonCrypto's operations directly in the `CryptoInputStream` implementation - since `CryptoInputStream` is already tightly coupled with Swift's CommonCrypto, so IMHO there's nothing that [CryptoStream](https://github.com/pubnub/swift/blob/master/Sources/PubNub/Helpers/Crypto/CryptoStream.swift)  abstracts away. 

I can try to use CCCOperations directly in `CryptoInputStream`, but I'm not sure yet what would be the scope of changes after introducing this idea.